userspace/engine/ruleset.cpp

```diff
@@ -145,25 +141,17 @@ void falco_ruleset::ruleset_filters::evttypes_for_ruleset(std::set<uint16_t> &ev
 {
 	evttypes.clear();
 
-	for(auto &wrap : m_filters)
+	for(auto &filter : m_filters)
 	{
-		auto fevttypes = wrap->evttypes();
-		evttypes.insert(fevttypes.begin(), fevttypes.end());
+		evttypes.insert(filter->evttypes.begin(), filter->evttypes.end());
 	}
 }
 
-void falco_ruleset::add(string &source,
-			string &name,
-			set<string> &tags,
-			std::shared_ptr<gen_event_filter> filter)
+void falco_ruleset::add(falco_filter& filter)
```

Comment:
Why not pass a shared_ptr<falco_filter> instead? It's still shared_ptrs within ruleset.cpp and the first thing add() does is make a copy into a shared pointer.